marshal.go

```diff
@@ -145,19 +148,103 @@ func structToEntries(v reflect.Value, schema bool, opt *marshalOptions) (entries
 				entries = append(entries, &Entry{Block: block})
 			}
 
-		case tag.optional && field.v.IsZero() && !schema:
-
 		default:
 			attr, err := fieldToAttr(field, tag, schema)
 			if err != nil {
 				return nil, nil, err
 			}
+			if tag.optional && !schema {
+				equalsDefault, err := valueEqualsDefault(field.v, attr)
```

Comment:
I just dawned on me that I think this function can just be simplified to `attr.Value.String() == attr.DefaultValue.String()`

Reply:
map type might have a problem with its undetermined key order

Reply:
Order should be stable. If it's not, that's a bug!

Reply:
Stable when stringing I mean.

Reply:
do we need to ensure the order in the `String()` function in Value? it's not at the moment. value and default value comes from 2 sources,